src/LuaTransformer.ts

```diff
@@ -209,6 +203,34 @@ export class LuaTransformer {
         }
     }
 
+    public transformDeclareStatement(node: ts.Statement): StatementVisitResult {
```

Comment:
IMO we shouldn't do this anymore. First, it would be quite a surprising behavior if some code would work in .d.ts  files (since we don't transform them) and won't in regular. Second, it's pretty much valid to declare it, because it also may be used to augment _G type.

Reply:
I see your point about the inconsistency being confusing, but I don't see how that can be a valid declaration. Such variables could be declared as a property of `_G`, but this error wouldn't be triggered by that.

Also I think it would be a good idea to leave this function in (just returning undefined) as it makes sense as an api method.

Reply:
The best type for `_G` is `typeof globalThis`, which includes all globally declared symbols